File: Go/HappyNumber.go
package main

import "fmt"
import "sort"  
import "math" 

func happy(n int) bool {
	m := make(map[int]bool)
	for n > 1 {
		m[n] = true
		var x int
		// for ( int i = n, j = 0; )
		for x, n = n, 0; x > 0; x /= 10 {
			d := x % 10
			n += d * d
		}
		if m[n] {
			return false
		}
	}
	return true
}

func findNorm(num int) float64 {
  sum := num * num
  m := 0
  for num != 1 {
    m = 0
    for num > 0 {
      m += (num % 10) * (num % 10)
      num /= 10
    }
    num = m
    sum += num * num
  }
  var norm float64 = math.Sqrt(float64(sum))
  return norm
}


func main() {
    var arg1 int
    var arg2 int
    fmt.Println("Enter Your First Argument: ")
    fmt.Scanln(&arg1)
    fmt.Println("Enter Your Second Argument: ")
    fmt.Scanln(&arg2)
    
    
    var temp int = 0
    if(arg1 > arg2){
        temp = arg1
        arg1 = arg2
        arg2 = temp
    }
     m := make(map[float64]int)
    for num := arg1; num <=arg2; num++ {
        if(happy(num)){
            m[findNorm(num)]= num
        }
    }
    if(len(m)==0){
        fmt.Println("NO BODY IS HAPPY!")
    }else{
          count := 0
          keys := make([]float64, 0, len(m))
          for k := range m{
            keys = append(keys, k)
          }
          sort.Sort(sort.Reverse(sort.Float64Slice(keys)))
        
          for _, k := range keys {
            fmt.Println(m[k])
            count++
            if(count >9){
                break;
            }
        }

    }
        
    }

File: Python/HappyNumber.py
import math
import operator

def happy(n):
    past = set()			
    while n != 1:
        n = sum(int(i)**2 for i in str(n))
        if n in past:
            return False
        past.add(n)
    return True
    
def FindNorm(n):
      norm = n**2
      m=0
      while(n!=1):
          m=0
          while(n>0):
              m += (n%10)**2
              n //= 10
          n = m
          norm += n**2     
      norm = math.sqrt(norm)
      return norm
            
     
 
def main():
    arg1 = int(input("Enter the first argument: "))
    arg2 = int(input("Enter the second argument: "))
    
    print("\nFirst Argument: " + str(arg1))
    print("Second Argument: " + str(arg2))
    if arg1>arg2:
        temp = arg1
        arg1 = arg2
        arg2 = temp

    print("\nFirst Argument: " + str(arg1))
    print("Second Argument: " + str(arg2))
   
    dicList = {}
    for i in range(arg1, arg2 + 1):
        if happy(i):
            dicList[FindNorm(i)] = i
        
    if(len(dicList) == 0):
        print("NO BODY IS HAPPY!")
    else:
        count =0
        for j in sorted(dicList.keys(),reverse=True):
            print(dicList[j])
            count+=1
            if(count>9):
                break	
        
main()
 
 
 
 
 
